paymentsheet/src/main/java/com/stripe/android/paymentsheet/elements/common/TextFieldElement.kt

```diff
@@ -0,0 +1,108 @@
+package com.stripe.android.paymentsheet.elements.common
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.distinctUntilChanged
+import com.stripe.android.paymentsheet.NotNullMutableLiveData
+import com.stripe.android.paymentsheet.R
+
+/**
+ * This class will provide the onValueChanged and onFocusChanged functionality to the element's
+ * composable.  These functions will update the observables as needed.  It is responsible for
+ * exposing immutable observers for its data
+ */
+internal open class TextFieldElement(private val textFieldConfig: TextFieldConfigInterface) {
+    val debugLabel = textFieldConfig.debugLabel
+    private val isDebug = true
+
+    /** This is all the information that can be observed on the element */
+    private val _input: NotNullMutableLiveData<String> = NotNullMutableLiveData("")
+    val input: LiveData<String> = _input.distinctUntilChanged()
```

Comment:
I highly recommend using `MutableStateFlow` instead of `MutableLiveData` - there's really no reason to use `MutableLiveData` anymore.

```
private val _input = MutableStateFlow("")
val input: LiveData<String> = _input.asLiveData().distinctUntilChanged()
```

Reply:
I think we discussed pushing the removal of LiveData to another PR.

Reply:
That's fair. In this case, it seemed preferable to use `MutableLiveData` instead of introducing `NotNullMutableLiveData` which is only used in a few places.

Reply:
I will post a review with the change when this one is complete.